addons/website_sale/models/product.py

```diff
@@ -430,6 +430,14 @@ class Product(models.Model):
 
     product_variant_image_ids = fields.One2many('product.image', 'product_variant_id', string="Extra Variant Images")
 
+    website_url = fields.Char('Website URL', compute='_compute_product_website_url', help='The full URL to access the document through the website.')
```

Comment:
I think you should use the related mixin to add the `website_url` field, unless you had a reason to do it this way ? See with @rdeodoo if you don't know.

Reply:
+1

Reply:
Actually since product.product inherits from product.template, it already has the mixin.
At the end, website_url from product.product and product.template would be the one from the mixin (as expected) but there is no easy way to have a different value for product and template.
At first I was hoping we could do a condition check on record._name to return different compute value if template or product, but given the inherit**s** mechanism, record will always be product.template even when called on product.product.

Adding the mixin is thus not a solution either since it already (indirectly) has the mixin.

This is the best way I can think of.

@JKE-be What do you think?